App.tsx

```diff
@@ -45,23 +49,55 @@ export default function App() {
 
     const [error, setError] = React.useState('');
     const [date, setDate] = React.useState(new Date());
+    const [showDatePicker, setShowDatePicker] = React.useState(false);
     const [timings, setTimings] = React.useState<Timings>();
 
-    React.useEffect(() => {
-        (async () => {
+    const fetchTimings = async (selectedDate: Date) => {
+        try {
             let { status } = await Location.requestForegroundPermissionsAsync();
             if (status !== 'granted') {
                 setError('Location permission denied.');
                 return;
             }
 
             let location = await Location.getCurrentPositionAsync({});
-            const timings = await (await fetch(`http://api.aladhan.com/v1/calendar/${date.getFullYear()}/${date.getMonth() + 1}?latitude=${location.coords.latitude}&longitude=${location.coords.longitude}&method=${METHOD}&tune=${TUNE}`)).json();
+            const timings = await fetch(
+                `http://api.aladhan.com/v1/calendar/${selectedDate.getFullYear()}/${selectedDate.getMonth() + 1}?latitude=${location.coords.latitude}&longitude=${location.coords.longitude}&method=${METHOD}&tune=${TUNE}`
+            ).then((res) => res.json());
+
+            setTimings(timings.data[selectedDate.getDate() - 1].timings);
+        } catch (error) {
+            setError('Error fetching timings.');
+        }
+    };
 
-            setTimings(timings.data[new Date().getDate() - 1].timings);
-        })();
+    React.useEffect(() => {
+        fetchTimings(date);
```

Comment:
Are we using the fetchTimings function anywhere else as well? If not, we can just keep all of that code here directly instead of creating a function.

Reply:
Yes, I incorporated it in the useEffect itself.